gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/configuration/application/registration/ClientRegistrationMapper.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.rest.api.service.impl.configuration.application.registration;
+
+import io.gravitee.rest.api.model.NewApplicationEntity;
+import io.gravitee.rest.api.model.UpdateApplicationEntity;
+import io.gravitee.rest.api.service.impl.configuration.application.registration.client.register.ClientRegistrationRequest;
+import org.mapstruct.BeanMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface ClientRegistrationMapper {
+    ClientRegistrationMapper INSTANCE = Mappers.getMapper(ClientRegistrationMapper.class);
+
+    @Mapping(source = "name", target = "clientName")
+    @Mapping(source = "settings.OAuthClient.applicationType", target = "applicationType")
+    @Mapping(source = "settings.OAuthClient.clientUri", target = "clientUri")
+    @Mapping(source = "settings.OAuthClient.grantTypes", target = "grantTypes")
+    @Mapping(source = "settings.OAuthClient.logoUri", target = "logoUri")
+    @Mapping(source = "settings.OAuthClient.redirectUris", target = "redirectUris")
+    @Mapping(source = "settings.OAuthClient.responseTypes", target = "responseTypes")
+    ClientRegistrationRequest toClientRegistrationRequest(NewApplicationEntity application);
+
+    @BeanMapping(nullValuePropertyMappingStrategy = org.mapstruct.NullValuePropertyMappingStrategy.IGNORE, ignoreByDefault = true)
+    @Mapping(source = "name", target = "clientName")
+    @Mapping(source = "settings.OAuthClient.applicationType", target = "applicationType")
+    @Mapping(source = "settings.OAuthClient.clientUri", target = "clientUri")
+    @Mapping(source = "settings.OAuthClient.grantTypes", target = "grantTypes")
+    @Mapping(source = "settings.OAuthClient.logoUri", target = "logoUri")
+    @Mapping(source = "settings.OAuthClient.redirectUris", target = "redirectUris")
+    @Mapping(source = "settings.OAuthClient.responseTypes", target = "responseTypes")
+    void toClientRegistrationRequest(@MappingTarget ClientRegistrationRequest target, UpdateApplicationEntity application);
```

Comment:
why use `@MappingTarget` and not return it instead of void ?

Reply:
Because this object is partially pre-filled with ClientRegistrationResponse data